### DISCUSSION TITLE
Assistance with Interpreting eBus Traffic Using readallvaillantregisters.sh

### DISCUSSION THREAD

**Discussion by mediendiver** [john30/ebusd#1119] [General]
My heat pump from Vaillant has been running for several days now. I retrieve the Ebus traffic using the eBus Adapter 5 over WLAN, which also works:

Bei mir läuft seit einigen Tagen meine Wärmepumpe von Vaillant. Den Ebus Verkehr rufe ich mit dem eBus Adapter 5 über WLAN ab. Was auch funktioniert:

`version: ebusd 23.2.23.2
update check: version 23.3 available, device firmware 1[3c17] available
device: 192.168.178.122:9999, enhanced, firmware 1.1[3c11].1[3c11]
access: *
signal: acquired
symbol rate: 88
max symbol rate: 175
min arbitration micros: 7
max arbitration micros: 1028
min symbol latency: 0
max symbol latency: 60
scan: finished
reconnects: 0
masters: 4
messages: 714
conditional: 0
poll: 1
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0902;HW=5103", loaded "vaillant/08.hmu00.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV3;SW=0709;HW=3704", loaded "vaillant/15.ctlv3.csv"
address 26: slave, scanned "MF=Vaillant;ID=VR_71;SW=0203;HW=5904", loaded "vaillant/26.vr_71.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0902;HW=5103", loaded "vaillant/76.vwzio.csv"
address ec: slave, scanned "MF=Vaillant;ID=SOL00;SW=0709;HW=3704", loaded "vaillant/ec.sol.sc.csv"
`
Now I'm dealing with the script readallvaillantregisters.sh. If I enter, for example, this command, I get the following result:

Jetzt beschäftige ich mich mit dem Skript readallvaillantregisters.sh. Wenn ich z.B. diesen Befehl eingebe bekomme ich folgendes Ergebnis:
`./readallvaillantregisters.sh -a 26
0 = 020407
1 = 02fc06
2 = 020007
3 = 02ff0f
4 = 02ff0f
5 = 020100
6 = 020100
7 = 020100
8 = 02ff0f
9 = 02ff0f
10 = 02ff0f
11 = 02ff0f
12 = 02ff0f
13 = 0100
14 = 0102
15 = 0100
16 = 020000
17 = 020000
18 = 0100
19 = 0100
20 = 0100
21 = 0101
22 = 0100
23 = 0100
24 = 0100
25 = 0100
26 = 0100
27 = 0100
28 = 0100
29 = 0100
30 = 0101
31 = 0100
32 = 0101
33 = 02e803
34 = 02e803
35 = 0101
36 = 0100
37 = 0100
38 = 0100
39 = 0100
40 = 0100
41 = 03373715
42 = 031e0c17
43 = 0101
44 = 0101
45 = 0100
46 = 0100
47 = 0100
48 = 0affffffffffffffffffff
49 = 0100
50 = 0100
51 = 0100
52 = 0100
53 = 01b7
54 = 0100
55 = 0100
56 = 0100
57 = 0100
58 = 0100
59 = 0100
60 = 0100
61 = 0100
62 = 0415020000
63 = 0100
64 = 0100
65 = 0100
66 = 08ffffffffffffffff
67 = 0100
68 = 020000
69 = 020000
70 = 03a10100
71 = 03a30100
72 = 03a20100
73 = 030080aa
74 = 030080aa
75 = 03008000
76 = 03008000
77 = 03008000
78 = 030080aa
79 = 030080aa
80 = 030080aa
81 = 030080aa
82 = 030080aa
83 = 0100
84 = 0100
85 = 0100
86 = 0100
87 = 0100
88 = 0112
89 = 0116
90 = 011e
91 = 010c
92 = 0142
93 = 02f400
94 = 02b400
95 = 022401
96 = 028c00
97 = 026c00
98 = 029401
99 = 0100
100 = 0100
101 = 0100
102 = 0100
103 = 0100
104 = 0100
105 = 0135
106 = 0100
107 = 021700
108 = 010c
109 = 0100
110 = 0100
111 = 0100
112 = 0101
113 = 0100
114 = 027801
115 = 02ffff
116 = 0100
117 = 0100
118 = 00
119 = 00
120 = ERR: read timeout
121 = ERR: read timeout
122 = 00
123 = 00
124 = 00
125 = 00
126 = 00
127 = 00
128 = 00
129 = 00
130 = 00
131 = 00
132 = 00
133 = 00
134 = 00
135 = 00
136 = 00
137 = 00
138 = 00
139 = 00
140 = 00
141 = 00
142 = 00
143 = ERR: read timeout
144 = 00
145 = 00
146 = 00
147 = 00
148 = 00
149 = 00
150 = 00
151 = 00
152 = 00
153 = 00
154 = 00
155 = 00
156 = 00
157 = 00
158 = 00
159 = 00
160 = 00
161 = 00
162 = 00
163 = 00
164 = 00
165 = 00
166 = 00
167 = 00
168 = 00
169 = 00
170 = 00
171 = 00
172 = 00
173 = 00
174 = 00
175 = 00
176 = 00
177 = 00
178 = 00
179 = 00
180 = 00
181 = 00
182 = ERR: read timeout
183 = ERR: read timeout
184 = 00
185 = 00
186 = 00
187 = 00
188 = 00
189 = 00
190 = ERR: read timeout
191 = ERR: read timeout
192 = ERR: read timeout
193 = 00
194 = 00
195 = 00
196 = 00
197 = 00
198 = 00
199 = 00
200 = 00
201 = 00
202 = 00
203 = 00
204 = 00
205 = 00
206 = 00
207 = 00
208 = 00
209 = 00
210 = 00
211 = 00
212 = ERR: read timeout
213 = 00
214 = 00
215 = 00
216 = 00
217 = 00
218 = 00
219 = 00
220 = 00
221 = 00
222 = 00
223 = 00
224 = 00
225 = 00
226 = 00
227 = 00
228 = 00
229 = 00
230 = 00
231 = 00
232 = 00
233 = 00
234 = 00
235 = 00
236 = 00
237 = 00
238 = 00
239 = 00
240 = 00
241 = 00
242 = 00
243 = 00
244 = 00
245 = 00
246 = 00
247 = 00
248 = 00
249 = 00
250 = 00
251 = 00
252 = ERR: read timeout
253 = 00
254 = 00
255 = 00
256 = 00
257 = 00
258 = 00
259 = 00
260 = 00
261 = 00
262 = 00
263 = 00
264 = 00
265 = 00
266 = 00
267 = ERR: read timeout
268 = 00
269 = 00
270 = 00
271 = 00
272 = 00
273 = ERR: read timeout
274 = 00
275 = 00
276 = 00
277 = 00
278 = 00
279 = 00
280 = 00
281 = 00
282 = 00
283 = 00
284 = ERR: read timeout
285 = ERR: read timeout
286 = ERR: read timeout
287 = ERR: read timeout
288 = ERR: read timeout
289 = 00
290 = ERR: read timeout
291 = ERR: read timeout
292 = ERR: read timeout
293 = ERR: read timeout
294 = ERR: SYN received
295 = ERR: read timeout
296 = ERR: read timeout
297 = ERR: read timeout
298 = ERR: read timeout
299 = ERR: read timeout
300 = ERR: read timeout
301 = ERR: read timeout
302 = ERR: read timeout
303 = ERR: read timeout
304 = ERR: read timeout
305 = ERR: read timeout
306 = ERR: read timeout
307 = ERR: read timeout
308 = ERR: read timeout
309 = ERR: read timeout
310 = ERR: read timeout
311 = ERR: read timeout
312 = ERR: read timeout
313 = ERR: read timeout
314 = ERR: read timeout
315 = ERR: read timeout
316 = ERR: read timeout
317 = ERR: read timeout
318 = ERR: read timeout
319 = ERR: read timeout
320 = ERR: read timeout
321 = ERR: read timeout
322 = ERR: read timeout
323 = ERR: read timeout
324 = ERR: read timeout
325 = ERR: read timeout
326 = ERR: read timeout
327 = ERR: SYN received
328 = 00
329 = 00
330 = 00
331 = 00
332 = 00
333 = 00
334 = 00
335 = 00
336 = 00
337 = ERR: read timeout
338 = 00
339 = 00
340 = 00
341 = 00
342 = 00
343 = 00
344 = 00
345 = 00
346 = 00
347 = 00
348 = 00
349 = 00
350 = 00
351 = 00
352 = 00
353 = ERR: read timeout
354 = 00
355 = 00
356 = 00
357 = 00
358 = 00
359 = 00
360 = 00
361 = 00
362 = 00
363 = 00
364 = 00
365 = 00
366 = 00
367 = 00
368 = 00
369 = 00
370 = 00
371 = 00
372 = 00
373 = 00
374 = 00
375 = 00
376 = 00
377 = 00
378 = 00
379 = 00
380 = 00
381 = 00
382 = 00
383 = 00
384 = 00
385 = 00
386 = 00
387 = 00
388 = 00
389 = 00
390 = ERR: read timeout
391 = ERR: read timeout
392 = ERR: read timeout
393 = ERR: read timeout
394 = ERR: read timeout
395 = ERR: read timeout
396 = ERR: read timeout
397 = ERR: read timeout
398 = ERR: read timeout
399 = ERR: read timeout
400 = ERR: read timeout
401 = ERR: read timeout
402 = ERR: read timeout
403 = ERR: read timeout
404 = ERR: read timeout
405 = ERR: read timeout
406 = ERR: read timeout
407 = ERR: read timeout
408 = ERR: wrong symbol received
409 = 00
410 = 00
411 = 00
412 = 00
413 = ERR: read timeout
414 = 00
415 = 00
416 = 00
417 = 00
418 = ERR: read timeout
419 = 00
420 = 00
421 = 00
422 = 00
423 = ERR: read timeout
424 = 00
425 = 00
426 = 00
427 = 00
428 = 00
429 = 00
430 = ERR: read timeout
431 = 00
432 = 00
433 = 00
434 = ERR: SYN received
435 = ERR: read timeout
436 = ERR: read timeout
437 = ERR: read timeout
438 = ERR: read timeout
439 = 00
440 = ERR: read timeout
441 = 00
442 = ERR: read timeout
443 = 00
444 = ERR: read timeout
445 = 00
446 = 00
447 = 00
448 = 00
449 = 00
450 = 00
451 = ERR: read timeout
452 = 00
453 = 00
454 = 00
455 = 00
456 = 00
457 = 00
458 = 00
459 = 00
460 = ERR: read timeout
461 = 00
462 = 00
463 = 00
464 = 00
465 = 00
466 = 00
467 = 00
468 = 00
469 = 00
470 = 00
471 = 00
472 = 00
473 = 00
474 = 00
475 = 00
476 = 00
477 = 00
478 = 00
479 = 00
480 = 00
481 = 00
482 = 00
483 = 00
484 = 00
485 = 00
486 = 00
487 = 00
488 = 00
489 = 00
490 = 00
491 = 00
492 = ERR: read timeout
493 = ERR: read timeout
494 = 00
495 = ERR: read timeout
496 = ERR: read timeout
497 = ERR: read timeout
498 = 00
499 = 00
500 = 00
501 = 00
502 = 00
503 = ERR: read timeout
504 = ERR: read timeout
505 = 00
506 = 0114
507 = 0101
508 = 010c
509 = 040000ae01
510 = 00
511 = 00`
I have already looked at the documentation 'Creating new message definition files' in the wiki, but unfortunately, I don't understand it. What exactly can I do with the result now, and how do I convert it into a CSV file? I'm just not making any progress with this. Thank you in advance for any assistance.

Ich habe mir jetzt schon die Dokumantion Creating new message definition files im Wiki angeschaut, aber leider verstehe ich das nicht. Was genau kann ich jetzt mit dem Ergebnis anfangen, wie setze ich das jetzt in eine csv Datei um? Da komme ich jetzt einfach nicht weiter. 
Dankeschön.

**Comment by cybersmart-eu** [comment]
Da sind wir schon zu zweit die feststellen, dass das fsnz schön komplex ist - bei mir mit Ochsner Wärmepumpe für dienich nir ein passendes CSV konfigurieren darf - die vorhandenen passen nur teilweise.
Aber für vaillant gibt es doch schon extrem viele Konfigurationen - kommen da im default keine aussagekräftige Werte bei dir an?

VG
Uwe

**Reply by mediendiver** [reply]
Ja, ich bekomme Werte von den csv Dateien. Allerdings möchte ich doch gerne mittels des Auslesens meiner Geräte bessere csv Dateien erstellen, sodass ich doch mehr auslesen kann. Aber wie schon geschrieben, komme ich da nicht weiter.

**Reply by cybersmart-eu** [reply]
Das ist die Fleissarbeit - ich sitze heute schon den ganzen Tag und optimiere meine Ochsner CSV, dokumentiere mir alles was ich so rausfinde um
mehr über die Daten und den Aufbau der CSV und Templates zu lernen. Ausserdem dann Suchen nach Werten die mir noch fehlen, aber am Regler angezeigt werden und somit auch irgendwo auf dem Bis zu finden sind … Fleissarbeit

**Reply by mediendiver** [reply]
Würde ich ja auch machen, nur habe ich keinen Plan, was ich mit dem Ergebnis vom Befehl readallvaillantregisters.sh anfangen soll/kann.
